cms/nextjs-contentful-store/pages/[countryCode]/[lang]/[product].tsx

```diff
@@ -0,0 +1,192 @@
+import _ from "lodash";
+import React, { useState, useEffect } from "react";
+import { GetStaticPaths, GetStaticProps } from "next";
+import Image from "next/image";
+import Link from "next/link";
+import { Price, PricesContainer, AddToCartButton } from "@commercelayer/react-components";
+import { useGetToken } from "@hooks/GetToken";
+import locale from "@locale/index";
+import Page from "@components/Page";
+import { Country, Product } from "@typings/models";
+import { parseImg, parseLanguageCode } from "@utils/parser";
+import contentfulApi from "@utils/contentful/api";
+
+type Props = {
+  lang: string;
+  countries: Country[];
+  country: Country | undefined;
+  product: Product;
+  buildLanguages: Country[];
+};
+
+const ProductPage: React.FC<Props> = ({ lang, countries, country, product, buildLanguages }) => {
+  const countryCode = country?.code.toLowerCase() as string;
+  const clMarketId = country?.marketId as string;
+  const clEndpoint = process.env.NEXT_PUBLIC_CL_ENDPOINT as string;
+  const clToken = useGetToken({
+    scope: clMarketId,
+    countryCode: countryCode
+  });
+  const languageCode = parseLanguageCode(lang, "toLowerCase", true);
+
+  const imgUrl = parseImg(_.first(product?.images)?.url as string);
+  const firstVariantCode = _.first(product?.variants)?.code as string;
+  const variantOptions = product?.variants?.map((variant) => {
+    return {
+      label: variant.size?.name,
+      code: variant.code,
+      lineItem: {
+        name: product.name,
+        imageUrl: _.first(variant.images)?.url
+      }
+    };
+  });
+
+  const [selectedVariant, setSelectedVariant] = useState<string>();
+
+  useEffect(() => {
+    setSelectedVariant(firstVariantCode);
+  }, [firstVariantCode]);
+
+  return !lang || !product ? null : (
```

Comment:
Given the type definition, I'm not expecting `lang` and `product` to be not defined.

```suggestion
  return (
```

Reply:
Given the use of `fallback: true` in `getStaticPaths` and that `lang` and `product` are fetched from the router `params` query in `getStaticProps`, they can be undefined at initial render (without transition from `next/link`) or full reload. I have instead used `fallback: "blocking"` to hang initial render till the data is ready instead of using a loading state as agreed. Grazie.